### PR TITLE
Remove folders and collections

### PR DESCRIPTION
**Pull request by dariusk**
I believe I've completely removed folders and collections from the code base. I took care to do each step as an atomic commit, so I recommend reviewing commit-by-commit in chronological order to see what was removed, since they are in easy-to-understand functional chunks that way.

The remaining major task is to remove folder-related integration and unit tests.

References: CV2-2827

## Type of change

- [X] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Automated test (add or update automated tests)

## How has this been tested?

Planning to majorly overhaul integration tests especially here.

## Checklist

- [X] I have performed a self-review of my own code
- [ ] I've made sure my branch is runnable and given good testing steps in the PR description
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [X] I have removed the /* eslint-disable @calm/react-intl/missing-attribute */ from any files I've worked on and added a `description` prop to all `<FormattedMessage />` components that were missing it.
- [X] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

**Comment by amoedoamorim** [comment]
Reviewing this and it's looking great! Please let us know when it leaves WIP state and it's ready for a final review!

**Comment by dariusk** [comment]
@caiosba @amoedoamorim  Not quite done but I want to draw special attention to adb71879f3c836e46ada25a031577315471bf09e -- this was not specified in the ticket but came up when I was searching for references to "Folder". The commit message explains what I'm doing but basically I'm getting rid of folder-related actions from `ProjectActions`, which itself is a generic thing for doing actions on a given item the user is looking at. I believe everything I removed is safe, but in particular there is some permissions-related stuff removed around whether a user has permission to view a folder that I want extra eyes on.

I am also curious about the `Can` here -- what does `create Project` mean anymore as a permission? https://github.com/meedan/check-web/pull/1563/commits/adb71879f3c836e46ada25a031577315471bf09e#diff-e79d2aeaa2d1b40dfa17b8cabf2da2892149719f79a51be47f35a7ee384a4929R155

We should probably also rename `ProjectActions`. The only place it's used now is `search/SavedSearch.js`, which is a list, so maybe `ListActions`?

**Comment by dariusk** [comment]
@amoedoamorim Okay! I believe I've completely removed folders and collections from the code base. I took care to do each step as an atomic commit, so I recommend reviewing commit-by-commit in chronological order to see what was removed, since they are in easy-to-understand functional chunks that way.

**Comment by caiosba** [comment]
> @caiosba @amoedoamorim Not quite done but I want to draw special attention to [adb7187](https://github.com/meedan/check-web/commit/adb71879f3c836e46ada25a031577315471bf09e) -- this was not specified in the ticket but came up when I was searching for references to "Folder". The commit message explains what I'm doing but basically I'm getting rid of folder-related actions from `ProjectActions`, which itself is a generic thing for doing actions on a given item the user is looking at. I believe everything I removed is safe, but in particular there is some permissions-related stuff removed around whether a user has permission to view a folder that I want extra eyes on.
> 
> I am also curious about the `Can` here -- what does `create Project` mean anymore as a permission? [adb7187#diff-e79d2aeaa2d1b40dfa17b8cabf2da2892149719f79a51be47f35a7ee384a4929R155](https://github.com/meedan/check-web/commit/adb71879f3c836e46ada25a031577315471bf09e#diff-e79d2aeaa2d1b40dfa17b8cabf2da2892149719f79a51be47f35a7ee384a4929R155)
> 
> We should probably also rename `ProjectActions`. The only place it's used now is `search/SavedSearch.js`, which is a list, so maybe `ListActions`?

@dariusk , the commit https://github.com/meedan/check-web/commit/adb71879f3c836e46ada25a031577315471bf09e looks good at a first glance. I also checked `ProjectActions.js` in your branch and it looks sane. But I'm going to try once you're done. If I understood it correctly, `ProjectActions.js` is really `ListActions.js` for now, right? In that case, I guess we can replace `<Can permission="create Project">` by `<Can permission="create SavedSearch">`?

**Comment by dariusk** [comment]
@caiosba Great! And yeah `ProjectActions` is basically `ListActions` yes. Also if `create SavedSearch` is a valid permission then for sure that's what is needed here. Thanks!

File: localization/react-intl/src/app/components/project/Project.json
Comment (amoedoamorim) [inline localization/react-intl/src/app/components/project/Project.json:1]: Looks like this whole file can be deleted too?

File: localization/react-intl/src/app/components/project/ProjectGroup.json
Comment (amoedoamorim) [inline localization/react-intl/src/app/components/project/ProjectGroup.json:1]: Looks like this whole file can be deleted too?

File: src/app/components/Root.js [hunk 1]
```diff
@@ -82,25 +80,23 @@ class Root extends Component {
                   <Route path=":team" component={Team} />
                   <Route path=":team/settings(/:tab)" component={Team} />
                   <Route path=":team/media/:mediaId" component={MediaPage} />
-                  <Route path=":team/project/:projectId/media/:mediaId" component={MediaPage} />
+                  <Redirect from=":team/project/:projectId/media/:mediaId" to=":team/media/:mediaId" />
```
Comment (amoedoamorim) [inline src/app/components/Root.js:83]: Nice! I didn't know about this Redirect component!

File: localization/react-intl/src/app/components/project/ProjectGroup.json
Comment (dariusk) [reply localization/react-intl/src/app/components/project/ProjectGroup.json:1]: I think maybe every file in `localization/react-intl/src/app/components/project/` can be deleted? I'm searching the source for the ids being used anywhere and can't find them

**Review by amoedoamorim — APPROVED** [review]

**Review by amoedoamorim — APPROVED** [review]
LGTM!

**Comment by dariusk** [comment]
@caiosba I believe 46b3cf9c08119a5cdbfb9f06a452dcc95ad07598 handles the "redirect on login to folder" logic -- now when you log in it just checks for if you have a team and sends you there, otherwise it sends you to `/check/me/workspaces`

**Comment by caiosba** [comment]
> @caiosba I believe [46b3cf9](https://github.com/meedan/check-web/commit/46b3cf9c08119a5cdbfb9f06a452dcc95ad07598) handles the "redirect on login to folder" logic -- now when you log in it just checks for if you have a team and sends you there, otherwise it sends you to `/check/me/workspaces`

Sounds great, thank you!

**Comment by dariusk** [comment]
@caiosba I *think* I have squashed all the integration test bugs. The [remaining failing tests](https://app.travis-ci.com/github/meedan/check-web/jobs/607838857) are all search-related so I'm wondering if it's an elasticsearch issue or something??

**Comment by caiosba** [comment]
> @caiosba I *think* I have squashed all the integration test bugs. The [remaining failing tests](https://app.travis-ci.com/github/meedan/check-web/jobs/607838857) are all search-related so I'm wondering if it's an elasticsearch issue or something??

Amazing work @dariusk ! Are they failing consistently? Maybe they are just flaking. In any case, I noticed that `develop` failed too, so I just restarted them.

**Comment by dariusk** [comment]
Merging this since the failing tests are known flakes.

**Pull request merged**